sklearn/linear_model/_logistic.py

```diff
@@ -1428,7 +1557,7 @@ def __init__(self, penalty='l2', dual=False, tol=1e-4, C=1.0,
                  fit_intercept=True, intercept_scaling=1, class_weight=None,
                  random_state=None, solver='lbfgs', max_iter=100,
                  multi_class='auto', verbose=0, warm_start=False, n_jobs=None,
-                 l1_ratio=None):
+                 l1_ratio=None, precondition=True):
```

Comment:
The docstring of the estimators should be updated to document this parameter. In particular it should explain that this is a diagonal preconditioning: solving the problem on the center and scaled data and then transforming the coef / intercept to recover the solution on the original learning problem.

It should explain the difference with preprocessing the data, in particular in terms of regularization.